### ISSUE TITLE
Error shutting down the node - BlockStoreQueue

### ISSUE THREAD

**Issue by mikedennis** [stratisproject/StratisBitcoinFullNode#2630]
Running on stratis testnet with fairly recent build (about a day old)

```
[2018-10-30 03:16:25.8249 7] INFO: Stratis.Bitcoin.Base.BaseFeature.Dispose Disposing partial validator.
[2018-10-30 03:16:25.9733 7] ERROR: Stratis.Bitcoin.Builder.FullNodeFeatureExecutor.Execute An error occurred: 'System.AggregateException: One or more errors occurred. (Can't add block to pending!) ---> System.Exception: Can't add block to pending!
   at Stratis.Bitcoin.Features.BlockStore.BlockStoreQueue.AddToPending(ChainedHeaderBlock chainedHeaderBlock) in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin.Features.BlockStore\BlockStoreQueue.cs:line 303
   at Stratis.Bitcoin.Features.BlockStore.BlockStoreSignaled.OnNextCore(ChainedHeaderBlock blockPair) in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin.Features.BlockStore\BlockStoreSignaled.cs:line 88
   at System.Reactive.SafeObserver`1.OnNext(TSource value) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\Internal\SafeObserver.cs:line 41
   at System.Reactive.Observer`1.OnNext(T value) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\Internal\Observers.cs:line 80
   at System.Reactive.SynchronizedObserver`1.OnNextCore(T value) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\Internal\SynchronizedObserver.cs:line 20
   at Stratis.Bitcoin.Signals.Signaler`1.Broadcast(T item) in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin\Signals\Signaler.cs:line 70
   at Stratis.Bitcoin.Signals.Signals.SignalBlockConnected(ChainedHeaderBlock chainedHeaderBlock) in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin\Signals\Signals.cs:line 94
   at Stratis.Bitcoin.Consensus.ConsensusManager.ConnectChainAsync(List`1 blocksToConnect, Boolean blockMined) in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin\Consensus\ConsensusManager.cs:line 680
   at Stratis.Bitcoin.Consensus.ConsensusManager.FullyValidateLockedAsync(ChainedHeader newTip, Boolean blockMined) in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin\Consensus\ConsensusManager.cs:line 542
   at Stratis.Bitcoin.Consensus.ConsensusManager.OnPartialValidationSucceededAsync(ChainedHeader chainedHeader) in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin\Consensus\ConsensusManager.cs:line 412
   at Stratis.Bitcoin.Consensus.ConsensusManager.OnPartialValidationCompletedCallbackAsync(ValidationContext validationContext) in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin\Consensus\ConsensusManager.cs:line 354
   at Stratis.Bitcoin.Consensus.Validators.PartialValidator.OnEnqueueAsync(PartialValidationItem item, CancellationToken cancellationtoken) in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin\Consensus\Validators\BlockValidator.cs:line 151
   at Stratis.Bitcoin.Utilities.AsyncQueue`1.ConsumerAsync() in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin\Utilities\AsyncQueue.cs:line 135
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Stratis.Bitcoin.Utilities.AsyncQueue`1.DisposeInternal(Boolean calledFromConsumerTask) in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin\Utilities\AsyncQueue.cs:line 248
   at Stratis.Bitcoin.Utilities.AsyncQueue`1.Dispose() in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin\Utilities\AsyncQueue.cs:line 234
   at Stratis.Bitcoin.Consensus.Validators.PartialValidator.Dispose() in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin\Consensus\Validators\BlockValidator.cs:line 142
   at Stratis.Bitcoin.Base.BaseFeature.Dispose() in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin\Base\BaseFeature.cs:line 294
   at Stratis.Bitcoin.Builder.FullNodeFeatureExecutor.<>c.<Dispose>b__4_0(IFullNodeFeature feature) in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin\Builder\FullNodeFeatureExecutor.cs:line 67
   at Stratis.Bitcoin.Builder.FullNodeFeatureExecutor.Execute(Action`1 callback, Boolean reverseOrder) in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin\Builder\FullNodeFeatureExecutor.cs:line 102
---> (Inner Exception #0) System.Exception: Can't add block to pending!
   at Stratis.Bitcoin.Features.BlockStore.BlockStoreQueue.AddToPending(ChainedHeaderBlock chainedHeaderBlock) in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin.Features.BlockStore\BlockStoreQueue.cs:line 303
   at Stratis.Bitcoin.Features.BlockStore.BlockStoreSignaled.OnNextCore(ChainedHeaderBlock blockPair) in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin.Features.BlockStore\BlockStoreSignaled.cs:line 88
   at System.Reactive.SafeObserver`1.OnNext(TSource value) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\Internal\SafeObserver.cs:line 41
   at System.Reactive.Observer`1.OnNext(T value) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\Internal\Observers.cs:line 80
   at System.Reactive.SynchronizedObserver`1.OnNextCore(T value) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\Internal\SynchronizedObserver.cs:line 20
   at Stratis.Bitcoin.Signals.Signaler`1.Broadcast(T item) in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin\Signals\Signaler.cs:line 70
   at Stratis.Bitcoin.Signals.Signals.SignalBlockConnected(ChainedHeaderBlock chainedHeaderBlock) in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin\Signals\Signals.cs:line 94
   at Stratis.Bitcoin.Consensus.ConsensusManager.ConnectChainAsync(List`1 blocksToConnect, Boolean blockMined) in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin\Consensus\ConsensusManager.cs:line 680
   at Stratis.Bitcoin.Consensus.ConsensusManager.FullyValidateLockedAsync(ChainedHeader newTip, Boolean blockMined) in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin\Consensus\ConsensusManager.cs:line 542
   at Stratis.Bitcoin.Consensus.ConsensusManager.OnPartialValidationSucceededAsync(ChainedHeader chainedHeader) in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin\Consensus\ConsensusManager.cs:line 412
   at Stratis.Bitcoin.Consensus.ConsensusManager.OnPartialValidationCompletedCallbackAsync(ValidationContext validationContext) in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin\Consensus\ConsensusManager.cs:line 354
   at Stratis.Bitcoin.Consensus.Validators.PartialValidator.OnEnqueueAsync(PartialValidationItem item, CancellationToken cancellationtoken) in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin\Consensus\Validators\BlockValidator.cs:line 151
   at Stratis.Bitcoin.Utilities.AsyncQueue`1.ConsumerAsync() in C:\Users\fullnodemike\source\StratisBitcoinFullNode\src\Stratis.Bitcoin\Utilities\AsyncQueue.cs:line 135<---
'
[2018-10-30 03:16:25.9733 7] ERROR: Stratis.Bitcoin.Builder.FullNodeFeatureExecutor.Dispose An error occurred stopping the application.

**Comment by fassadlr** [comment]
@mikedennis I have a PR that fixes this #2627

**Comment by fassadlr** [comment]
Closed with #2627
